genie-web/src/main/java/com/netflix/genie/web/jpa/services/JpaJobPersistenceServiceImpl.java

```diff
@@ -602,10 +583,48 @@ public void updateJobStatus(
      * {@inheritDoc}
      */
     @Override
+    @Transactional(readOnly = true)
+    public boolean isV4(
+        @NotBlank(message = "Id is missing and is required") final String id
+    ) {
+        log.debug("Read v4 flag from db for job {} ", id);
+        return this.jobRepository
+            .findByUniqueId(id, IsV4JobProjection.class)
+            .orElseThrow(
+                () -> {
+                    final String errorMessage = "No job with id " + id + "exists. Unable to get v4 flag.";
+                    log.error(errorMessage);
+                    return new GenieJobNotFoundException(errorMessage);
+                }
+            ).isV4();
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    @Transactional(readOnly = true)
     public Optional<JobStatus> getJobStatus(@NotBlank(message = "Job id is missing and is required") final String id) {
+        // TODO: Need to decide on consistency of API. Do we throw exception? Return Optional? Runtime or Checked?
+        //       getJobArchiveLocation throws exception cause the underlying return type is already optional. So
+        //       do we return Optional<Optional<String>> that's really weird
         return this.jobRepository.findByUniqueId(id, JobStatusProjection.class).map(JobStatusProjection::getStatus);
```

Comment:
The other two here are throwing `GenieNotFoundException`. What's the reason for not doing so for this one?

Reply:
At least in this file, everything else is throwing `GenieJobNotFoundException`.
I know we need some consistency and a strategy at project level, but everything in this file is behaving the same,  I think having a single method behave differently for no apparent reason is more confusing, no?